crates/wasi-provider/src/wasi_runtime.rs

```diff
@@ -294,29 +311,53 @@ impl WasiRuntime {
                 Err(e) => {
                     let message = "unable to run module";
                     error!("{}: {:?}", message, e);
-                    status_sender
-                        .broadcast(Status::Terminated {
+                    send(
+                        status_sender.clone(),
+                        name.clone(),
+                        Status::Terminated {
                             failed: true,
                             message: message.into(),
                             timestamp: chrono::Utc::now(),
-                        })
-                        .expect("status should be able to send");
+                        },
+                        &mut cx,
+                    );
                     return Err(anyhow::anyhow!("{}: {}", message, e));
                 }
             };
 
             info!("module run complete");
-            status_sender
-                .broadcast(Status::Terminated {
+            send(
+                status_sender.clone(),
+                name,
+                Status::Terminated {
                     failed: false,
                     message: "Module run completed".into(),
                     timestamp: chrono::Utc::now(),
-                })
-                .expect("status should be able to send");
+                },
+                &mut cx,
+            );
             Ok(())
         });
         // Wait for the interrupt to be sent back to us
         let interrupt = rx.await?;
         Ok((interrupt, handle))
     }
 }
+
+fn send(mut sender: Sender<(String, Status)>, name: String, status: Status, cx: &mut Context<'_>) {
```

Comment:
Sorry, to reopen this conversation, but it seems that the use of `tokio::Sender` is defined in the `ModuleRunContext` of this crate. I had changed the channel in `Pod::queue`, yes, but I dont think that means we have to use it everywhere. It resulted in cleaner code in `Pod::queue`, but clearly not here. I dont like the fact that `tokio::sync::watch` drops messages, but I think it would be ok here.

Reply:
Because we are using a shared channel for all containers, we can't use watch or we'll end up dropping status events if there are multiple containers